src/platform/packages/private/kbn-grid-layout/grid/grid_row/grid_row.tsx

```diff
@@ -133,8 +131,23 @@ export const GridRow = forwardRef<HTMLDivElement, GridRowProps>(
             }
           });
 
+        /**
```

Comment:
This replaces the action we execute currently on drop:
```
setPanelIdsInOrder(getKeysInOrder(gridLayoutStateManager.gridLayout$.getValue()[rowIndex].panels));
```

Although it might seem like overkill for now, it's necessary for future keyboard navigation support, particularly to handle the cancel behavior. For example, when the user activates keyboard mode, moves a panel, and wants to cancel, we'll need to track the last stable grid layout to allow the action to be undone.

Reply:
Ooooh, I haven't noticed we have this functionality. Nice! 41322eb525908f4818859c57a83875a28c8d7eb7 and c9bfa0d9d367cbf28c7ba33ae03a4097d54d2777 

I renamed the variables with your proposal - I also had a problem with finding the right name, was thinking about `activeGridLayout` or `dragGridLayout` for not-stable state, but nothing felt good tbh.

Reply:
Actually, now when I changed the names with your suggestion @Heenawter  I am thinking I should work a little more on it - only have a `gridLayout$` during anyting that happens in the app and have a `proposedGridLayout$` filled with data _only_ when dragging/resizing. This way I don't have to worry about keeping them in sync outside of dragging/resizing action. I'll implement this change tomorrow, but just letting you know it's coming 💃🏼

Reply:
This sounds like a really good solution!! :) I think we could then remove a lot of our "is there an interaction event happening" checks, because we can be **very** explicit on whether we are subscribing to the "tentative" layout or the "finalized" layout 🎉

Reply:
This is implemented already! Not as clear as I was imagining (with all the `proposedGridLayout ?? gridLayout` checks, but still better than before. If you have some better ideas, unless it's a quick fix, I'd rather address it separately to not blow out this PR 🙏🏼